src/axolotl/utils/quantization.py

```diff
@@ -45,50 +53,59 @@ def get_ptq_config(
-            return UIntXWeightOnlyConfig(
-                dtype=weight_dtype.value,
-                group_size=group_size,
-                set_inductor_config=False,
-            )
-        if weight_dtype == TorchIntDType.int8:
-            if group_size is None:
-                raise ValueError(
-                    "group_size must be specified for int8 weight only quantization"
-                )
-            return Int8WeightOnlyConfig(
-                group_size=group_size,
-            )
-        if weight_dtype == TorchIntDType.int4:
-            if group_size is None:
-                raise ValueError(
-                    "group_size must be specified for int4 weight only quantization"
-                )
-            return Int4WeightOnlyConfig(
-                group_size=group_size,
-            )
-    if activation_dtype == TorchIntDType.int4 and weight_dtype == TorchIntDType.int4:
-        return Int4DynamicActivationInt4WeightConfig()
-    if activation_dtype == TorchIntDType.int8 and weight_dtype == TorchIntDType.int8:
-        return Int8DynamicActivationInt8WeightConfig()
-    if activation_dtype == TorchIntDType.int8 and weight_dtype == TorchIntDType.int4:
-        return Int8DynamicActivationInt4WeightConfig()
+        if weight_dtype == TorchAOQuantDType.int8:
+            raise ValueError("Int8WeightOnlyConfig is not supported by torchao QAT.")
+        if weight_dtype == TorchAOQuantDType.int4:
+            return Int4WeightOnlyConfig(group_size=group_size or -1, version=2)
+    if (
+        activation_dtype == TorchAOQuantDType.int4
+        and weight_dtype == TorchAOQuantDType.int4
+    ):
+        raise ValueError(
+            "Int4DynamicActivationInt4WeightConfig is not supported by torchao QAT."
+        )
+    if (
+        activation_dtype == TorchAOQuantDType.int8
+        and weight_dtype == TorchAOQuantDType.int8
+    ):
+        raise ValueError(
+            "Int8DynamicActivationInt8WeightConfig is not supported by torchao QAT."
+        )
+    if (
+        activation_dtype == TorchAOQuantDType.int8
+        and weight_dtype == TorchAOQuantDType.int4
+    ):
+        return Int8DynamicActivationInt4WeightConfig(group_size=group_size or -1)
+    if (
+        activation_dtype == TorchAOQuantDType.float8_e4m3fn
+        and weight_dtype == TorchAOQuantDType.float8_e4m3fn
+    ):
+        return Float8DynamicActivationFloat8WeightConfig(version=2)
+    if (
+        activation_dtype == TorchAOQuantDType.float8_e4m3fn
+        and weight_dtype == TorchAOQuantDType.int4
+    ):
+        return Float8DynamicActivationInt4WeightConfig()
+    if weight_dtype == TorchAOQuantDType.nvfp4:
+        from torchao.prototype.mx_formats import NVFP4InferenceConfig
+
+        if group_size is not None and group_size != 16:
+            raise ValueError("NVFP4 quantization must use a group_size of 16")
+        return NVFP4InferenceConfig()
```

Comment:
note that this one actually only does 1 level of scaling (unlike the specification in NVIDIA, which does 2 levels). This is because the PTQ only supports 1 level of scaling currently, but we plan to change this in the future (i.e. add dynamic per tensor scale). QAT supports both but currently uses 1 level of scaling to match the PTQ config